[GTK][WPE] Update some sysprof marks

carlosgcampos opened the pull request:
#### d7da21e4b2df769387cedcc3475994dc66b056d6
<pre>
[GTK][WPE] Update some sysprof marks
<a href="https://bugs.webkit.org/show_bug.cgi?id=303982">https://bugs.webkit.org/show_bug.cgi?id=303982</a>

Reviewed by Nikolas Zimmermann.

This patch includes the following changes in sysprof marks:

 - Rename UpdateRendering as LayerTreeHostRenderingUpdate since we
   previously had UpdateRendering and RenderingUpdate. This is
   consistent with existing RenderingUpdate but makint it clear this is
   the one in LayerTreeHost.

 - Rename ScheduleRenderingUpdate as LayerTreeHostScheduleRenderingUpdate
   for the same reason. In this case the mark was duplicated.

 - Remove WaitPaintingCompletion since we no longer wait for tiles
   synchronously except for forced repaints.

 - Convert RenderingUpdateRunLoopObserverFired to a single mark, since
   we don&apos;t really need to know its duration that is already covered by
   LayerTreeHostRenderingUpdate mark when updateRendering() is called.

 - Add ScheduleComposition mark with information about the compositor state.

 - Update FrameComplete mark to include information about the compositor state.

Canonical link: <a href="https://commits.webkit.org/304289@main">https://commits.webkit.org/304289@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/35de27f11f4def2371237b5b54bb11d9fc897978

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135152 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7544 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46411 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142658 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/86942 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/137021 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8179 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7393 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103275 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/86942 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138098 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5821 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121126 "Found 1 new API test failure: TestWebKitAPI.WebKit.MSEIsPlayingAudio (failure)") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84131 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5616 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/3227 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/3252 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/127155 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/114829 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39301 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145356 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/133640 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7228 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39869 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111650 "Found 1 new test failure: fast/scrolling/rtl-scrollbars-sticky-document.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7271 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6053 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112015 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5456 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117420 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/61164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20844 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7281 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35566 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/166505 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7037 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/70833 "Built successfully") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/43525 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7258 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7139 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann commented on Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp:
```diff
@@ -411,8 +411,24 @@ void ThreadedCompositor::requestComposition(CompositionReason reason)
+
 void ThreadedCompositor::scheduleUpdateLocked()
 {
+    WTFEmitSignpost(this, ScheduleComposition, "reasons: %s, state: %s, waiting for tiles: %s, render timer active: %s", reasonsToString(m_state.reasons).ascii().data(), stateToString(m_state.state).characters(), m_state.isWaitingForTiles ? "yes" : "no", m_renderTimer.isActive() ? "yes" : "no");
```
How about adding an inlined boolToString helper, encapsulating the yes/no?

carlosgcampos replied:
Not sure it's worth it, and we would need to put it in a common place since we do this inn several places.

nikolaszimmermann reviewed (approved):
Thanks!

carlosgcampos committed:
[GTK][WPE] Update some sysprof marks
https://bugs.webkit.org/show_bug.cgi?id=303982

Reviewed by Nikolas Zimmermann.

This patch includes the following changes in sysprof marks:

 - Rename UpdateRendering as LayerTreeHostRenderingUpdate since we
   previously had UpdateRendering and RenderingUpdate. This is
   consistent with existing RenderingUpdate but makint it clear this is
   the one in LayerTreeHost.

 - Rename ScheduleRenderingUpdate as LayerTreeHostScheduleRenderingUpdate
   for the same reason. In this case the mark was duplicated.

 - Remove WaitPaintingCompletion since we no longer wait for tiles
   synchronously except for forced repaints.

 - Convert RenderingUpdateRunLoopObserverFired to a single mark, since
   we don't really need to know its duration that is already covered by
   LayerTreeHostRenderingUpdate mark when updateRendering() is called.

 - Add ScheduleComposition mark with information about the compositor state.

 - Update FrameComplete mark to include information about the compositor state.

Canonical link: https://commits.webkit.org/304289@main
